tests/test_start_date.py

```diff
@@ -201,33 +196,27 @@ def start_date_test(self):
                     raise Exception("Expectations are set incorrectly. {} cannot have a "
                                     "replication method of {}".format(stream, replication_type))
 
-
-                # Skip the remaining assertions for inventories since it is append only
-                if stream == 'inventories':
-                    continue
-
-
-                # BUG | https://stitchdata.atlassian.net/browse/SRCE-3681
-                # Skipping these two streams until BUG resolved.
-                # NOTE: we skip inventories ^ for a different reason leave that as is
-                if stream in {'roles', 'employees', 'locations'}: # TODO REMOVE
-                    continue
-
-
                 # Verify all data from first sync has bookmark values >= start_date .
-                records_from_sync_1 = set(row.get('data').get(comparison_key)
-                                          for row in synced_records_1.get(stream, []).get('messages', []))
-                for record in records_from_sync_1:
-                    self.assertGreaterEqual(self.parse_date(record), self.parse_date(self.START_DATE_1),
-                                            msg="Record was created prior to start date for 1st sync.\n" +
-                                            "Sync 1 start_date: {}\n".format(self.START_DATE_1) +
-                                            "Record bookmark: {} ".format(record))
-
-                # Verify all data from second sync has bookmark values >= start_date 2.
-                records_from_sync_2 = set(row.get('data').get(comparison_key)
-                                          for row in synced_records_2.get(stream, {}).get('messages', []))
-                for record in records_from_sync_2:
-                    self.assertGreaterEqual(self.parse_date(record), self.parse_date(self.START_DATE_2),
-                                            msg="Record was created prior to start date for 2nd sync.\n" +
-                                            "Sync 2 start_date: {}\n".format(self.START_DATE_2) +
-                                            "Record bookmark: {} ".format(record))
+                self.assertRecordsDateGreaterEqual(synced_records_1, stream, self.START_DATE_1)
+                self.assertRecordsDateGreaterEqual(synced_records_2, stream, self.START_DATE_2)
+
+    def assertRecordsDateGreaterEqual(self, sync_records, stream, start_date):
+        replication_type = self.expected_replication_method().get(stream)
```

Comment:
The replication_type is already defined on 187. I think it would actually make sense to move the start_date_key logic from below out of this assertion up to where replication type is defined at the start of the loop. And then just add it as a parameter to this method. Just a suggestion, not a huge deal either way.

Reply:
I pulled it into a base method since the logic for determine start date is also used elsewhere in the base. But I don't entirely agree. If it's not used anywhere else why pull it out of the method?

Reply:
Well this pattern of abstracting assertions into methods isn't one we typically use. I think it's fine as long as it is clear. I think it is more clear now.